2024-08-30 13:07:07,953 - ERROR - Exception on /Specs/M6274 -1 [PUT]
Traceback (most recent call last):
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask_cors\extension.py", line 178, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\Desktop\MaintTracker\Backend\app.py", line 108, in specs_update
    specs.qrcode = data.get('qrcode', specs.qrcode)
                                      ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'qrcode'
2024-08-30 13:08:24,632 - ERROR - Exception on /Specs [POST]
Traceback (most recent call last):
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask_cors\extension.py", line 178, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\Desktop\MaintTracker\Backend\app.py", line 90, in add_specs
    motor = request.json['motor']
            ~~~~~~~~~~~~^^^^^^^^^
KeyError: 'motor'
2024-08-30 13:10:09,413 - ERROR - Exception on /Specs/M6274 -1 [PUT]
Traceback (most recent call last):
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask_cors\extension.py", line 178, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\Desktop\MaintTracker\Backend\app.py", line 108, in specs_update
    specs.qrcode = data.get('qrcode', specs.qrcode)
                                      ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'qrcode'
2024-09-05 09:18:33,075 - DEBUG - New IBST added: <IBST (transient 2042809028144)>
2024-09-05 09:19:50,662 - DEBUG - New IBST added: <IBST (transient 2042809108672)>
2024-09-05 09:20:36,174 - DEBUG - New IBST added: <IBST (transient 2042809028144)>
2024-09-05 09:20:52,404 - DEBUG - New IBST added: <IBST (transient 2042809118560)>
2024-09-05 09:21:21,904 - DEBUG - New IBST added: <IBST (transient 2042808784736)>
2024-09-05 09:22:07,892 - DEBUG - New IBST added: <IBST (transient 2042809109872)>
2024-09-05 09:22:34,318 - DEBUG - New IBST added: <IBST (transient 2042808780320)>
2024-09-05 09:23:02,293 - DEBUG - New IBST added: <IBST (transient 2042809121248)>
2024-09-05 09:23:35,465 - DEBUG - New IBST added: <IBST (transient 2042809033616)>
2024-09-05 09:24:06,121 - DEBUG - New IBST added: <IBST (transient 2042808783440)>
2024-09-05 09:25:23,170 - DEBUG - New IBST added: <IBST (transient 2042809107280)>
2024-09-05 09:26:15,791 - DEBUG - New IBST added: <IBST (transient 2042809119568)>
2024-09-06 11:04:54,172 - DEBUG - New IBST added: <IBST (transient 2042809425920)>
2024-09-06 11:05:51,608 - DEBUG - New IBST added: <IBST (transient 2042808780992)>
2024-09-06 11:08:20,775 - DEBUG - New IBST added: <IBST (transient 2042809110880)>
2024-09-06 11:08:53,341 - DEBUG - New IBST added: <IBST (transient 2042809433888)>
2024-09-06 11:09:24,422 - DEBUG - New IBST added: <IBST (transient 2042809420160)>
2024-09-06 11:10:12,236 - DEBUG - New IBST added: <IBST (transient 2042809400592)>
2024-09-06 11:10:48,357 - DEBUG - New IBST added: <IBST (transient 2042809428272)>
2024-09-09 13:53:47,895 - ERROR - Exception on /Admin [POST]
Traceback (most recent call last):
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask_cors\extension.py", line 178, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\Desktop\MaintTracker\Backend\app.py", line 365, in add_admin
    id = request.json['id']
         ~~~~~~~~~~~~^^^^^^
TypeError: list indices must be integers or slices, not str
2024-09-09 14:18:17,320 - ERROR - Exception on /Admin [POST]
Traceback (most recent call last):
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask_cors\extension.py", line 178, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\.virtualenvs\Backend-NKSSbLU1\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Chris\Desktop\MaintTracker\Backend\app.py", line 366, in add_admin
    username = request.json['username']
               ~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'username'
2024-09-09 14:19:52,884 - DEBUG - New IBST added: <IBST (transient 2409339253584)>
2024-09-09 14:20:01,552 - DEBUG - New IBST added: <IBST (transient 2409339247008)>
2024-09-09 14:20:10,888 - DEBUG - New IBST added: <IBST (transient 2409339256512)>
2024-09-09 14:20:18,078 - DEBUG - New IBST added: <IBST (transient 2409339247344)>
2024-09-09 14:21:14,233 - DEBUG - New IBST added: <IBST (transient 2409339054288)>
2024-09-09 14:21:23,854 - DEBUG - New IBST added: <IBST (transient 2409339055968)>
2024-09-09 14:21:32,066 - DEBUG - New IBST added: <IBST (transient 2409339055104)>
2024-09-09 14:21:40,362 - DEBUG - New IBST added: <IBST (transient 2409339049632)>
2024-09-09 14:21:55,053 - DEBUG - New IBST added: <IBST (transient 2409339373232)>
2024-09-09 14:22:09,726 - DEBUG - New IBST added: <IBST (transient 2409339374816)>
2024-09-09 14:22:22,693 - DEBUG - New IBST added: <IBST (transient 2409339376496)>
2024-09-09 14:23:47,325 - DEBUG - New IBST added: <IBST (transient 2409339056448)>
